src/tests/unit_tests/test_util.py

```diff
@@ -146,3 +148,42 @@ def test_log_formatter_formats_correctly_2(logs_with_final_response):
         "\n-- Step: 2 -- \nFinal Response: \n Another final response."
     )
     assert log_formatter(logs_with_final_response) == expected_output
+
+
+@pytest.fixture
+def source_data():
+    return  " Cillum labore et culpa elit irure labore nostrud 12.45 minim cupidatat. Nulla nisi aliquip do duis elit tempor magna. Occaecat sunt nisi aliqua officia fugiat. Dolor ea ad mollit nulla ullamco sit voluptate cillum id laboris et proident anim. Culpa officia incididunt sit qui exercitation magna voluptate Lorem duis eu occaecat. Non occaecat deserunt voluptate cillum aliquip voluptate veniam. Ullamco commodo eiusmod consequat dolor cillum quis Lorem $45,000 labore tempor cupidatat  7 elit quis deserunt.  "
+
+@pytest.fixture
+def correct_result_data():
+    return "Labore deserunt 12.45 $45,000 ,7 sit velit nulla. Sint ipsum reprehenderit sint cupidatat amet est id anim exercitation fugiat adipisicing elit. Id est dolore minim magna occaecat aute. Est dolore culpa laborum non esse nostrud."
+
+@pytest.fixture
+def incorrect_result_data():
+    return "Labore deserunt 12.45 $45,000 ,7 ,56 , 65 sit velit nulla. Sint ipsum reprehenderit sint cupidatat amet est id anim exercitation fugiat adipisicing elit. Id est dolore minim magna occaecat aute. Est dolore culpa laborum non esse nostrud."
+
+
+def test_extract_numbers_from_text(source_data):
+    extracted_number = extract_numbers_from_text(source_data)
+    
+    #source data has these numbers in it
+    numbers_in_source_data = ['12.45', '45000','7']
+    if len(numbers_in_source_data) != len(extracted_number):
```

Comment:
I think `assert len(numbers_in_source_data) == len(extracted_number), "failed to extract a number"`  is clearer. And we no longer need the if-else statements